<launch>
<!--
	<arg name="node_start_delay" default="5" /> 
    <node name="launchpad_node" pkg="chefbot_bringup" type="launchpad_node.py" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@'" >
        <rosparam file="$(find chefbot_bringup)/param/board_config.yaml" command="load" />
    </node>
-->    
    <rosparam file="$(find chefbot_bringup)/config/controllers.yaml" command="load"/>
    <!-- change this next line to include the launch file for your robot's hardware drivers: -->
    <arg name="simulation" default="$(optenv TURTLEBOT_SIMULATION false)"/>
    <param name="/use_sim_time" value="$(arg simulation)"/>
    <arg name="urdf_file" default="$(find xacro)/xacro '$(find chefbot_description)/urdf/chefbot_base.urdf'" />
    <!--  <arg name="urdf_file" default="$(find xacro)/xacro.py '$(find arlobot_description)/urdf/common.urdf.xacro'" /> -->
    <param name="robot_description" command="$(arg urdf_file)" />
    <!-- important generally, but specifically utilised by the current app manager -->
    <!-- 
    <param name="robot/name" value="$(optenv ROBOT turtlebot)"/>
    <param name="robot/type" value="turtlebot"/>
    -->
    <rosparam param="base_width">0.186</rosparam>
    <rosparam param="ticks_meter">4504</rosparam>
    
    <!--
    <node name="lpid_velocity" pkg="chefbot_bringup" type="pid_velocity.py" output="screen">
        <remap from="wheel" to="lwheel"/>        
        <remap from="motor_cmd" to="left_wheel_speed"/>
        <remap from="wheel_vtarget" to="lwheel_vtarget"/>        
        <remap from="wheel_vel" to="lwheel_vel"/>
        <rosparam param="Kp">450</rosparam>
        <rosparam param="Ki">0.3</rosparam>
        <rosparam param="Kd">0</rosparam>
        <rosparam param="out_min">-255</rosparam>
        <rosparam param="out_max">255</rosparam>
        <rosparam param="rate">30</rosparam>
        <rosparam param="timeout_ticks">4</rosparam>
        <rosparam param="rolling_pts">5</rosparam>
    </node>
    <node name="rpid_velocity" pkg="chefbot_bringup" type="pid_velocity.py" output="screen">
        <remap from="wheel" to="rwheel"/>
        
        <remap from="motor_cmd" to="right_wheel_speed"/>
        <remap from="wheel_vtarget" to="rwheel_vtarget"/>
        
        <remap from="wheel_vel" to="rwheel_vel"/>
        <rosparam param="Kp">450</rosparam>
        <rosparam param="Ki">0.3</rosparam>
        <rosparam param="Kd">0</rosparam>
        <rosparam param="out_min">-255</rosparam>
        <rosparam param="out_max">255</rosparam>
        <rosparam param="rate">30</rosparam>
        <rosparam param="timeout_ticks">4</rosparam>
        <rosparam param="rolling_pts">5</rosparam>
    </node>
-->
    <group>
        <node name="robot_hardware_interface" pkg="chefbot_bringup" type="mbot" output="screen"/>
        <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher">
            <param name="publish_frequency" type="double" value="5.0" />
        </node>
        <node name="joint_vel_service" pkg="chefbot_bringup" type="joints_frm_pid_vel.py" output="screen"/>
        
        <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen"
		  args="
            roachbot/joint_state_controller
            roachbot/rwheel
			roachbot/lwheel
        "/>
    </group>
    <node pkg="chefbot_navig_cpp" type="twist_to_motor" name="twist_to_motor" output="screen">
        <remap from="rwheel_vtarget" to="/roachbot/rwheel/command"/>
        <remap from="lwheel_vtarget" to="/roachbot/lwheel/command"/>
    </node>
    <node pkg="chefbot_navig_cpp" type="diff_tf" name="diff_tf" output="screen"/>
</launch>
